### PR TITLE
chore: add batch call to get api/favorites

### PR DESCRIPTION
**Pull request by GmausDev**
I implement a batch call to the favorites call.

Now 
imagen

before 
imagen

PD: Im not to much into ts or angular , so maybe can be a better way to do this or pattern I mean usually i work as BackEnd in another languages and im not feel confident at all with this solution . Im open to do this in a different way.  Just here to improve and learn 📟 

it "solve " #167

### CODE REVIEW

File: packages/portal/src/app/shared/favorite-button/favorite-button/favorite-button.component.ts [hunk 1]
```diff
@@ -3,6 +3,10 @@ import { Component, Input, OnChanges, inject, signal } from "@angular/core";
 import { MatButtonModule } from "@angular/material/button";
 import { FavoriteService } from "../../favorite.service";
 import { UserService } from "../../user/user.service";
+import { Subject, of } from 'rxjs';
+import { debounceTime, switchMap } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
```
Comment (sinedied) [inline packages/portal/src/app/shared/favorite-button/favorite-button/favorite-button.component.ts:8]: Could you merge the imports here?

File: packages/portal/src/app/shared/favorite-button/favorite-button/favorite-button.component.ts [hunk 2]
```diff
@@ -11,24 +15,47 @@ import { UserService } from "../../user/user.service";
...
+        switchMap(async () => {
+          if (this.listing && this.user) {
+            return this.favoriteService.getFavorite(this.listing, this.user);
+          }
+          return of(null);
+        })
+      )
+      .subscribe((result) => {
+        if (result !== null && this.listing) {
+          console.log(result);
```
Comment (sinedied) [inline packages/portal/src/app/shared/favorite-button/favorite-button/favorite-button.component.ts:48]: Remove console.log

**Review by sinedied — CHANGES_REQUESTED** [review]
Thanks for the PR!
Though this doesn't actually implement batching here, this debounces the call when the favorite button is clicked or initially loaded.

The initial solution using `ngOnChanges` is flawed as during the bootstrap phase of the page it's called multiple times, the debounce addition feels to me a bit like a bandaid on top of it, adding more complexity to a component that shouldn't bear this complexity in the first place.

I think that it might be better to move all the loading logic outside of this component entirely to implement proper batching in the parent component, but I'm ok with having this as a temporary fix if the few comments I put are fixed.

**Comment by GmausDev** [comment]
Hi @sinedied  As I say im not was confident at all , tss is not my expertise 📎. I suppose there is a better solutions because as you say the `ngOnChanges` do his magic but im not understand at all how to work on this , so I googled solutions. There is no more time and I have been raised this on the discussion this Chore and maybe someone bring a better solution.

I´ll consider this like a small patch and expect a refactor :p 

Thank you for let me improve 😊

**Review by sinedied — DISMISSED** [review]

**Review by anfibiacreativa — CHANGES_REQUESTED** [review]
Please approve formatting fixes

**Review by GmausDev — COMMENTED** [review]
👌

Hi @anfibiacreativa I'm looking , is @manekinekko who approves this?

**Comment by GmausDev** [comment]
Umm , its done?
@anfibiacreativa  AzureDevOps its easier xD

**Comment by anfibiacreativa** [comment]
> 👌
> 
> Hi @anfibiacreativa I'm looking , is @manekinekko who approves this?

Current maintainers of the repo are @manekinekko, @sinedied and myself.

**Review by anfibiacreativa — APPROVED** [review]

**Pull request merged**